pkg/typesystem/typesystem_test.go

```diff
@@ -434,6 +434,125 @@ func TestInvalidRewriteValidations(t *testing.T) {
 			},
 			err: ErrRelationUndefined,
 		},
+		{
+			name: "Fails_If_Using_This_As_Relation_Name",
```

Comment:
Mind switching these tests over to use the DSL string instead? That's a continuous improvement initiative we've been trying to work on. Anytime we can update the tests to use the DSL strings instead would be nice 👍 .

Reply:
IIRC the DSL string will fail parsing.